inc/namespace.php

```diff
@@ -503,6 +507,52 @@ function load_plugins() {
 	}
 }
 
+/**
+ * Ensure Tachyon URL is using the current site hostname.
+ *
+ * @param string $tachyon_url The current tachyon URL.
+ * @return string The updated Tachyon URL.
+ */
+function set_tachyon_hostname( string $tachyon_url ) : string {
+	$tachyon_host = wp_parse_url( $tachyon_url, PHP_URL_HOST );
+	$current_host = wp_parse_url( site_url(), PHP_URL_HOST );
+
+	// Only do the replacement if the host name is not a subdomain of the Tachyon host.
+	if ( strpos( $current_host, $tachyon_host ) === false ) {
+		return str_replace( $tachyon_host, $current_host, $tachyon_url );
+	}
+
+	return $tachyon_url;
+}
+
+/**
+ * Ensure the S3 Uploads Bucket URL matches the current site hostname.
+ *
+ * @param array $dirs Uploads directories array.
+ * @return array
+ */
+function set_s3_uploads_bucket_url_hostname( array $dirs ) : array {
+	$s3_uploads = S3_Uploads::get_instance();
+
+	$primary_host = wp_parse_url( get_site_url( get_main_site_id( get_main_network_id() ) ), PHP_URL_HOST );
```

Comment:
We're using this here and in `TACHYON_URL` above, can we abstract given it's a somewhat complex thing? Also should both have error handling?